salt/modules/keystoneng.py

```diff
@@ -0,0 +1,382 @@
+import salt.utils
+
+HAS_SHADE = False
+try:
+    import shade
+    from shade.exc import OpenStackCloudException
+    HAS_SHADE = True
+except ImportError:
+    pass
+
+
+def __virtual__():
+    if HAS_SHADE:
+        return 'keystoneng'
+    return False
+
+
+__opts__ = {}
```

Comment:
Is this added for the keystone tests?  I don't think this line should be needed.

Reply:
probably copy/paste garbage. will remove